### PR TITLE
bugfix: pydantic model dynamic generation 

### PR DESCRIPTION
**Pull request by rbiseck3**
### Description
Fix how pydantic base model was being generated given a valid schema. Unit test added.

### CODE REVIEW

File: test/test_schema.py [hunk 1]
```diff
@@ -290,3 +291,29 @@ def fn(q: InputC) -> None:
...
+        f: list[float] = None,
+    ) -> None:
+        pass
+
+    class ExpectedInputModel(BaseModel):
+        a: int
+        b: Union[float, int] = 4
+        c: Optional[str] = "my_string"
+        d: bool = False
+        e: dict = None
```
Comment (jakub-sandomierz-deepsense-ai) [inline test/test_schema.py:312]: The main issue is that type subscription is lost. This creates discrepancy against the output of `/schema` endpoint, which shows full typing.

My proposal is to accept this solution and create a Github issue (or any other trackable item) which will be addressed later.

File: unstructured_platform_plugins/schema/json_schema.py [hunk 2]
```diff
@@ -212,25 +212,35 @@ def response_to_json_schema(return_annotation: Any) -> dict:
     return to_json_schema(val=return_annotation)
 
 
+def schema_to_base_model_type(t_string, name: str, type_info: dict) -> Type[BaseModel]:
```
Comment (jakub-sandomierz-deepsense-ai) [inline unstructured_platform_plugins/schema/json_schema.py:215]: `t_string` is not suggestive. I would propose changing it to something like `json_type_name: str`
```suggestion
def schema_to_base_model_type(json_type_name: str, name: str, type_info: dict) -> Type[BaseModel]:
```

File: unstructured_platform_plugins/schema/json_schema.py [hunk 3]
```diff
@@ -212,25 +212,35 @@ def response_to_json_schema(return_annotation: Any) -> dict:
...
-            if "default" in v:
-                input.append(v["default"])
-            else:
-                input.append(Ellipsis)
-            inputs[k] = tuple(input)
+    properties = schema.get("properties", {})
+    for k, v in properties.items():
+        if any_of := v.get("anyOf"):
+            type_info = [
+                schema_to_base_model_type(a["type"], name=f"{k}_{i}", type_info=a)
```
Comment (jakub-sandomierz-deepsense-ai) [inline unstructured_platform_plugins/schema/json_schema.py:231]: Please change name of variable `a` and `i` to be more descriptive

**Review by jakub-sandomierz-deepsense-ai — CHANGES_REQUESTED** [review]

File: test/test_schema.py [hunk 4]
```diff
@@ -290,3 +291,29 @@ def fn(q: InputC) -> None:
     }
     assert input_schema == expected_input_schema
     assert is_validate_dict(input_schema)
+
+
+def test_schema_to_base_model():
+    def fn(
+        a: int,
+        b: Union[float, int] = 4,
```
Comment (jakub-sandomierz-deepsense-ai) [inline test/test_schema.py:299]: New typing method e.g. `b: float | int = 4` is not supported and yields error. Underlying code expects union type of `GenericAlias`, but new typing is an direct instance of `UnionType`. Please see following example:
```
>>> from typing import Union
>>> new_typing = int | str
>>> old_typing = Union[int, str]
>>> new_typing == old_typing
True
>>> type(new_typing)
<class 'types.UnionType'>
>>> type(old_typing)
<class 'typing._UnionGenericAlias'>
```

**Review by jakub-sandomierz-deepsense-ai — CHANGES_REQUESTED** [review]
Found issue with new typing method

File: test/test_schema.py (hunk 4 above)
Comment (rbiseck3) [reply test/test_schema.py]: Good catch, this is now fixes. Haven't used this feature of 3.12 yet so forgot it existed.

File: test/test_schema.py (hunk 1 above)
Comment (rbiseck3) [reply test/test_schema.py:312]: > type subscription is lost

What do you mean by this?

Comment (jakub-sandomierz-deepsense-ai) [reply test/test_schema.py:312]: For example `e: dict[str, Any]` in `fn` is transformed into `e: dict` in `ExpectedInputModel`. Typing in brackets for collection types is omitted.

**Review by jakub-sandomierz-deepsense-ai — APPROVED** [review]

**Pull request merged**